components/script/body.rs

```diff
@@ -821,7 +821,16 @@ fn run_text_data_algorithm(bytes: Vec<u8>) -> Fallible<FetchedData> {
 
 #[allow(unsafe_code)]
 fn run_json_data_algorithm(cx: JSContext, bytes: Vec<u8>) -> Fallible<FetchedData> {
-    let json_text = String::from_utf8_lossy(&bytes);
+    // We first need to run https://encoding.spec.whatwg.org/#utf-8-decode
+    let mut cursor = 0;
+
+    if let Some(buffer) = bytes.get(cursor..cursor + 3) {
+        if buffer == [0xEF, 0xBB, 0xBF] {
+            cursor += 3;
+        }
+    }
+
```

Comment:
You could perhaps convert this to a string first and then do `trim_start_matches("\u{feff}")`. I think it makes sense to leave a comment explaining that `JS_ParseJSON` doesn't support parsing strings that include a BOM.

Reply:
There's also `slice::strip_prefix` (https://doc.rust-lang.org/std/primitive.slice.html#method.strip_prefix)

That could be used as `buffer.strip_prefix(&[0xEF, 0xBB, 0xBF]).unwrap_or(buffer)`.

Reply:
Thank you both 🙇

It looks like we already do this in [XMLHttpRequest](https://github.com/servo/servo/blob/main/components/script/dom/xmlhttprequest.rs#L1435) so I've made that shared code, and updated my comment to try to provide more context on why we are doing it.